include/caf/caf_overview.rst

```diff
@@ -0,0 +1,186 @@
+.. _caf_overview:
+
+Common Application Framework overview
+#####################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Common Application Framework (CAF) is a set of ready-for-use modules and events built on top of :ref:`event_manager`.
+Using CAF allows you to have a consistent event-based architecture in your application.
+
+An example of the application that is using the CAF modules in |NCS| is :ref:`nrf_desktop`.
+
+Architecture
+************
+
+In an event-based application, parts of the application functionality are separated into isolated modules that communicate with each other using Event Manager's events.
+These events are defined either by CAF or by the application.
+They are submitted by modules and other modules can subscribe and react to them.
+
+.. figure:: ../../doc/nrf/images/caf_overview.svg
+   :alt: Common Application Framework architecture example
+
+   Common Application Framework architecture example
+
+Every CAF module can use additional libraries or drivers.
+The drivers can be used to communicate with hardware.
+The module can then broadcast information from the driver using events (for example, the :ref:`buttons module <caf_buttons>` can submit a :c:struct:`button_event` on a button press).
+It can also forward information from application to the driver (for example, the :ref:`LEDs module <caf_leds>` can display a given LED effect upon receiving :c:struct:`led_event`).
+
+In your application, you can use any number of CAF modules together with your own custom modules.
+Custom modules can use both CAF and custom events.
+The figure shows an example application that uses one CAF module and one custom module, which has one custom event.
+
+Enabling CAF
+************
+
+To enable CAF, set the :option:`CONFIG_CAF` Kconfig option, initialize Event Manager, and submit the first :c:struct:`module_state_event`.
+You can then enable each :ref:`CAF module <lib_caf>` separately.
```

Comment:
I would remove `then` here. Modules are enabled using Kconfig options during build time. Event Manager is initialized when application is already running

Reply:
OK, I'm removing this line, as the modules are described later on, and this sentence goes a bit further than the section topic.